core/src/main/java/com/blink22/core/modules/example/ExampleModule.java

```diff
@@ -0,0 +1,20 @@
+package com.blink22.core.modules.example;
+
+import dagger.Module;
+import dagger.Provides;
+
+@Module
+public class ExampleModule {
```

Comment:
@mhemdan I believe the DI classes, e.g `Modules` & `Components`, need to be in a separate package rather than being distributed among different packages. I found many projects, [Google Todo](https://github.com/googlesamples/android-architecture/tree/todo-mvp-dagger), [Mindorks](https://github.com/MindorksOpenSource/android-mvp-architecture), and [Bourbon](https://github.com/hitherejoe/Bourbon) doing this. Check the attached screenshot from this [project](https://github.com/hitherejoe/Bourbon/tree/master/CoreCommon/src/main/java/com/hitherejoe/bourboncorecommon)
![screen shot 2018-09-27 at 3 49 38 pm](https://user-images.githubusercontent.com/8607770/46256133-e75c4800-c4a6-11e8-9f26-3ddb2055d553.png)